image.c

```diff
@@ -365,65 +340,52 @@ bool img_load_webp(const fileinfo_t *file, Imlib_Image *fframe, img_t *img)
-			img->multi.frames = emalloc(img->multi.cap * sizeof(img_frame_t));
-		} else if (info.frame_count > img->multi.cap) {
-			img->multi.cap = info.frame_count;
-			img->multi.frames = erealloc(img->multi.frames,
-			                             img->multi.cap * sizeof(img_frame_t));
-		}
+	/* Get global information for the image */
+	flags = WebPDemuxGetI(demux, WEBP_FF_FORMAT_FLAGS);
+	img->w = WebPDemuxGetI(demux, WEBP_FF_CANVAS_WIDTH);
+	img->h = WebPDemuxGetI(demux, WEBP_FF_CANVAS_HEIGHT);
 
-		/* Load and decode frames (also works on images with only 1 frame) */
-		img->multi.cnt = img->multi.sel = 0;
-		while (WebPAnimDecoderGetNext(dec, &buf, &ts)) {
-			im = imlib_create_image_using_copied_data(
-			     info.canvas_width, info.canvas_height, (DATA32*)buf);
-			imlib_context_set_image(im);
-			imlib_image_set_format("webp");
-			/* Get an iterator of this frame - used for frame info (duration, etc.) */
-			WebPDemuxGetFrame(demux, img->multi.cnt+1, &iter);
-			imlib_image_set_has_alpha((flags & ALPHA_FLAG) == ALPHA_FLAG);
-			/* Store info for this frame */
-			img->multi.frames[img->multi.cnt].im = im;
-			delay = iter.duration > 0 ? iter.duration : DEF_WEBP_DELAY;
-			img->multi.frames[img->multi.cnt].delay = delay;
-			img->multi.length += img->multi.frames[img->multi.cnt].delay;
-			img->multi.cnt++;
-		}
-		WebPDemuxReleaseIterator(&iter);
+	if (info.frame_count > img->multi.cap) {
+		img->multi.cap = info.frame_count;
+		img->multi.frames = erealloc(img->multi.frames,
+		                             img->multi.cap * sizeof(img_frame_t));
+	}
 
-		if (img->multi.cnt > 1) {
-			imlib_context_set_image(img->im);
-			imlib_free_image();
-			img->im = img->multi.frames[0].im;
-		} else if (img->multi.cnt == 1) {
-			imlib_context_set_image(img->multi.frames[0].im);
-			imlib_free_image();
-			img->multi.cnt = 0;
-		}
+	/* Load and decode frames (also works on images with only 1 frame) */
+	img->multi.cnt = img->multi.sel = 0;
+	while (WebPAnimDecoderGetNext(dec, &buf, &ts)) {
+		im = imlib_create_image_using_copied_data(
+		     info.canvas_width, info.canvas_height, (DATA32*)buf);
+		imlib_context_set_image(im);
+		imlib_image_set_format("webp");
+		/* Get an iterator of this frame - used for frame info (duration, etc.) */
+		WebPDemuxGetFrame(demux, img->multi.cnt+1, &iter);
+		imlib_image_set_has_alpha((flags & ALPHA_FLAG) == ALPHA_FLAG);
+		/* Store info for this frame */
+		img->multi.frames[img->multi.cnt].im = im;
+		delay = iter.duration > 0 ? iter.duration : DEF_WEBP_DELAY;
+		img->multi.frames[img->multi.cnt].delay = delay;
+		img->multi.length += img->multi.frames[img->multi.cnt].delay;
+		img->multi.cnt++;
+	}
+	WebPDemuxReleaseIterator(&iter);
+
+	if (img->multi.cnt > 1) {
 		imlib_context_set_image(img->im);
+		imlib_free_image();
+		img->im = img->multi.frames[0].im;
+	} else if (img->multi.cnt == 1) {
+		imlib_context_set_image(img->multi.frames[0].im);
+		imlib_free_image();
+		img->multi.cnt = 0;
```

Comment:
I think the gif loader has the same snippet. Could probably be turned into a separate function.